### PR TITLE
Only process the submodules of a module recursively, not its imports

### PR DESCRIPTION
**Pull request by JanJecmen**

### CODE REVIEW

File: src/StabilityCheck.jl [hunk 1]
```diff
@@ -52,20 +52,22 @@ include("annotations.jl")
 # Relies on `is_stable_function`.
 # (*) "all" can mean all or exported; cf. `SearchCfg`'s  `exported_names_only`.
 #
-is_stable_module(mod::Module, scfg :: SearchCfg = default_scfg) :: StCheckResults = begin
-    @debug "is_stable_module: $mod"
+const MODULES_SEEN = Set{Module}()
+is_stable_module_aux(mod::Module, root::Module, scfg :: SearchCfg = default_scfg) :: StCheckResults = begin
+    @debug "is_stable_module($mod)"
+    push!(MODULES_SEEN, mod)
```
Comment (ulysses4ever) [inline src/StabilityCheck.jl:58]: Presumably, `MODULES_SEEN` should be cleaned up in the top-level wrapper (`is_stable_module`)?

**Review by ulysses4ever — COMMENTED** [review]
Perhaps, one of your tests for recursive module processing could be modified to `import Pkg` (for example), and we could check that in the results we don't get anything from the imported module?

Could you remind me when modules-seen device is needed? Maybe it could have a simple test too?

**Comment by JanJecmen** [comment]
Yep, will add some tests.
The `MODULES_SEEN` was intended to prevent infinite recursion, eg. because a module's names list also the module itself. I now made it a local variable that is reset every time we enter through the top-level API which seems cleaner

File: test/runtests.jl [hunk 2]
```diff
@@ -171,9 +171,17 @@ end
 
 # Recursing into submodules
 
+# should not be included in the result
```
Comment (ulysses4ever) [inline test/runtests.jl:174]: It'd be nice to explain how it is tested exactly. I guess, it's the numbers in `AgStats`. But this is not very explicit. Best would be to take the result of `is_stable_module`, which is an array of method checks, so it has methods in it, and check that `ext` is not in that array. But at least, referring from the comment to `AgStats` could make it more clear that it is tested somehow...

File: test/runtests.jl [hunk 3]
```diff
@@ -224,9 +232,10 @@ module Nested
     unstable() = if rand()>0.5; 1; else ""; end
 end
 
-@testset "is_stable_module recursive     " begin
+@testset "is_stable_module nesting       " begin
     @test is_stable_moduleb(Nested, SearchCfg(exported_names_only=true))
     @test aggregateStats(is_stable_module(Nested)) == AgStats(13, 6, 0, 7, 0, 0, 0, 0, 0)
+    @test Nested.ext(21) == 42
```
Comment (ulysses4ever) [inline test/runtests.jl:238]: Mmm, why do we need this test?

**Review by ulysses4ever — APPROVED** [review]
Terrific, thanks!

**Comment by ulysses4ever** [comment]
@mergify squash

**Pull request merged**